src/app.py

```diff
@@ -194,6 +197,87 @@ def render_form(**kwargs):
                        new_url=new_url)
 
 
+@app.route("/pull_request_hook", methods=["POST"])
+def pull_request_hook():
+    event = request.headers.get('X-GitHub-Event')
+    if event == "ping":
+        return Response('Pong')
+    elif event != "pull_request":
+        return Response("Wrong event type", status=400)
+
+    payload = request.get_json()
+    action = payload['action']
+    if action not in ("opened", "edited", "reopened", "synchronize"):
+        return Response("Not Interested")
+    # status=204 : No Content
+
+    sender     = payload['sender']['login']
+
+    head_sha   = payload['pull_request']['head']['sha']
+    head_label = payload['pull_request']['head']['label']
+    head_ref   = payload['pull_request']['head']['ref']
+
+    base_sha   = payload['pull_request']['base']['sha']
+    base_label = payload['pull_request']['base']['label']
+    base_ref   = payload['pull_request']['base']['ref']
+
+    pull_num   = payload['pull_request']['number']
+    pull_url   = payload['pull_request']['html_url']
+
+    pull_ref   = "pull/{pull_num}/head".format(**locals())
+
+    # make sure data repo contains relevant commits
+    stdout, stderr, ret = fetch_data_ref(base_ref, pull_ref)
+
+    if ret == 0:
+        script = src_dir + "/tests/automerge_downtime_ok.py"
+        cmd = [script, base_sha, head_sha, sender]
+        stdout, stderr, ret = runcmd(cmd, cwd=global_data.topology_data_dir)
+
+    OK = "Yes" if ret == 0 else "No"
+
+    subject = "Pull Request {pull_url} {action}".format(**locals())
+
+    out = """\
+In Pull Request: {pull_url}
+GitHub User '{sender}' wants to merge branch {head_label}
+        (at commit {head_sha})
+into {base_label}
+        (at commit {base_sha})
+
+Eligible for downtime automerge? {OK}
+
+automerge_downtime script output:
+---
+{stdout}
+---
+{stderr}
+---
+""".format(**locals())
+
+    recipients = [ x + "@cs.wisc.edu" for x in ("edquist", "matyas", "blin") ]
```

Comment:
Can we leave a comment on the PR itself instead of emailing this set of people?

Reply:
I think we should do that once we're out of 'alpha testing mode'.

Reply:
> Can we leave a comment on the PR itself instead of emailing this set of people?

Just to note it here, we can post a comment on the PR with the [GitHub Issue Comments API](https://developer.github.com/v3/issues/comments/#create-a-comment).

Which is just to send a `POST /repos/:owner/:repo/issues/:number/comments` with post data `{"body": "comment text"}`.

I presume this requires github auth, which means deciding what user to use (do we have something like a service account/user for the `opensciencegrid` org?), and also means we have to decide how to store the auth token.  I'd guess we can create a separate account with minimal permissions, and store the auth token on disk rather than in the git repo.